src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchRunner.cs

```diff
@@ -0,0 +1,212 @@
+            CancellationToken cancellationToken)
+        {
+            return Task.Run(async () =>
+            {
+                var repository = Repository.Factory.GetCoreV3(source);
+                var resource = await repository.GetResourceAsync<PackageSearchResource>(cancellationToken);
+
+                if (resource == null)
+                {
+                    return null;
+                }
+
+                return await resource.SearchAsync(
+                    packageSearchArgs.SearchTerm,
+                    new SearchFilter(includePrerelease: packageSearchArgs.Prerelease),
+                    packageSearchArgs.Skip,
+                    packageSearchArgs.Take,
+                    packageSearchArgs.Logger,
+                    cancellationToken);
+            });
+        }
+
+        /// <summary>
+        /// Builds a task that perform exact match package search operation.
+        /// </summary>
+        /// <param name="source">A package source.</param>
+        /// <param name="packageSearchArgs">Package search arguments</param>
+        /// <param name="cancellationToken">A token to monitor for cancellation requests.</param>
+        /// <returns>A list of tasks that perform exact match package search operations.</returns>
+        private static Task<IEnumerable<IPackageSearchMetadata>> GetPackageAsync(
+            PackageSource source,
+            PackageSearchArgs packageSearchArgs,
+            CancellationToken cancellationToken)
+        {
+            return Task.Run(async () =>
+            {
+                using var cache = new SourceCacheContext();
+                var repository = Repository.Factory.GetCoreV3(source);
+                var resource = await repository.GetResourceAsync<PackageMetadataResource>(cancellationToken);
+
+                if (resource == null)
+                {
+                    return null;
+                }
+
+                return await resource.GetMetadataAsync(
+                    packageSearchArgs.SearchTerm,
+                    includePrerelease: packageSearchArgs.Prerelease,
+                    includeUnlisted: false,
+                    cache,
+                    packageSearchArgs.Logger,
+                    cancellationToken);
+            });
+        }
+
+        /// <summary>
+        /// Retrieves a list of package sources based on provided sources or default configuration sources.
+        /// </summary>
+        /// <param name="sources">The list of package sources provided.</param>
+        /// <param name="sourceProvider">The provider for package sources.</param>
+        /// <returns>A list of package sources.</returns>
+        private static IList<PackageSource> GetPackageSources(List<string> sources, IPackageSourceProvider sourceProvider)
+        {
+            List<PackageSource> configurationSources = sourceProvider.LoadPackageSources()
+                .Where(p => p.IsEnabled)
+                .ToList();
```

Comment:
your probably don't need this here. If there is a source param, you're gonna end up creating 2 lists.

You can just call ToList at line 165 once instead of 2 different places.

This is not critical here, as it's not gonna make or break the performance of the command, but it's a consideration for hot paths.